From: Jonathan Kang <jonathankang@gnome.org>
Date: Wed, 16 Mar 2022 11:15:34 +0800
Subject: network/connection-editor: always load all available VPN plugins

Previously, when a list of VPN plugins was loaded once, it will not load
again the next time you try to add a VPN connnection. This causes an
issue that if you keep gnome-control-center open and install a new VPN
plugin, the new VPN plugin won't show up in the VPN plugin list.

Fix that by always load all avilable VPN plugins when showing the
"Add VPN" dialog.

(cherry picked from commit ce93dbb3a0628ca611d91d01c2569d3c26204ff3)
---
 panels/network/connection-editor/vpn-helpers.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/panels/network/connection-editor/vpn-helpers.c b/panels/network/connection-editor/vpn-helpers.c
index 81ff4a2..e0358a9 100644
--- a/panels/network/connection-editor/vpn-helpers.c
+++ b/panels/network/connection-editor/vpn-helpers.c
@@ -53,14 +53,9 @@ _sort_vpn_plugins (NMVpnPluginInfo *aa, NMVpnPluginInfo *bb)
 GSList *
 vpn_get_plugins (void)
 {
-	static gboolean plugins_loaded = FALSE;
 	static GSList *plugins = NULL;
 	GSList *p;
 
-	if (G_LIKELY (plugins_loaded))
-		return plugins;
-	plugins_loaded = TRUE;
-
 	p = nm_vpn_plugin_info_list_load ();
 	plugins = NULL;
 	while (p) {
